package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -87,9 +79,8 @@ def layers_mapping(self):  # noqa: PLR0912
         self._layers_mapping = {}
 
         # Get datasets available in catalog
-        if IS_KEDRODATACATALOG and isinstance(self._catalog, KedroDataCatalog):
-            # Returns dataset names
-            datasets = self._catalog.list()
+        if hasattr(self._catalog, "keys") and callable(self._catalog.keys):
```

Comment:
Hey, I’m a bit confused, in some places we have to explicitly import KedroDataCatalog (I understand we had it for a few versions), but in others we don’t because it’s already handled somewhere else.
I know Ravi is away, but @ElenaKhaustova @merelcht if either of you has a moment to explain this, I’d really appreciate it!

Reply:
is this to pass mypy? because we only call it whereever we do 

`catalog: Union[DataCatalog, "KedroDataCatalog"]`